compiler/syntax/src/res_comments_table.ml

```diff
@@ -1508,7 +1552,87 @@ and walk_expression expr t comments =
         attach t.leading return_expr.pexp_loc leading;
         walk_expression return_expr t inside;
         attach t.trailing return_expr.pexp_loc trailing)
-  | _ -> ()
+  | Pexp_jsx_element
+      (Jsx_fragment
+         {
+           jsx_fragment_opening = opening_greater_than;
+           jsx_fragment_children = children;
+           jsx_fragment_closing = _closing_lesser_than;
+         }) ->
+    let opening_token = {expr.pexp_loc with loc_end = opening_greater_than} in
+    let on_same_line, rest = partition_by_on_same_line opening_token comments in
+    attach t.trailing opening_token on_same_line;
+    let exprs =
+      match children with
+      | Parsetree.JSXChildrenSpreading e -> [e]
+      | Parsetree.JSXChildrenItems xs -> xs
+    in
+    let xs = exprs |> List.map (fun e -> Expression e) in
+    walk_list xs t rest
+  | Pexp_jsx_element
+      (Jsx_unary_element
+         {jsx_unary_element_tag_name = tag; jsx_unary_element_props = []}) ->
+    let _, _, trailing = partition_by_loc comments tag.loc in
+    let after_expr, _ = partition_adjacent_trailing tag.loc trailing in
+    attach t.trailing tag.loc after_expr
+  | Pexp_jsx_element
+      (Jsx_unary_element
+         {jsx_unary_element_tag_name = tag; jsx_unary_element_props = props}) ->
+    let _leading, inside, trailing = partition_by_loc comments tag.loc in
+    walk_list
+      (props
+      |> List.filter_map (fun prop ->
+             match prop with
+             | Parsetree.JSXPropPunning (_, {loc}) ->
+               Some (ExprArgument {expr; loc})
+             | Parsetree.JSXPropValue ({loc}, _, expr) ->
+               let loc = {loc with loc_end = expr.pexp_loc.loc_end} in
+               Some (ExprArgument {expr; loc})
+             | Parsetree.JSXPropSpreading (loc, expr) ->
+               let loc = {loc with loc_end = expr.pexp_loc.loc_end} in
+               Some (ExprArgument {expr; loc})))
+      t trailing;
+    walk_expression expr t inside
+  | Pexp_jsx_element
+      (Jsx_container_element
+         {
+           jsx_container_element_tag_name_start = tag_name_start;
+           jsx_container_element_props = props;
+           jsx_container_element_opening_tag_end = opening_greater_than;
+           jsx_container_element_children = _children;
+         }) ->
+    let opening_greater_than_loc =
+      {
+        loc_start = opening_greater_than;
+        loc_end = opening_greater_than;
+        loc_ghost = false;
+      }
+    in
+    let after_opening_tag_name, _rest =
```

Comment:
Example:
```jsx
let _ = /* f1 */ <div /* f2 */ a=b /* f3 */ peek a boo ></div>
```

`f1` will at this point already be dealt with, as the value_binding will capture everything before and after the expression.

`f2` belongs to `div` because it is on the same line.

If `f2` were to be on the next line, I would attach it to the first prop instead.

```jsx
let _ = /* f1 */ <div 
   /* f2 */ a=b 
   /* f3 */ peek a boo></div>
```